homeassistant/components/duotecno/__init__.py

```diff
@@ -0,0 +1,39 @@
+"""The duotecno integration."""
+from __future__ import annotations
+
+from duotecno.controller import PyDuotecno
+from duotecno.exceptions import InvallidPassword, LoadFailure
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST, CONF_PASSWORD, CONF_PORT, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import PlatformNotReady
+
+from .const import DOMAIN
+
+PLATFORMS: list[Platform] = [Platform.SWITCH]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up duotecno from a config entry."""
+
+    hass.data.setdefault(DOMAIN, {})
+    controller = PyDuotecno()
+    hass.data[DOMAIN][entry.entry_id] = controller
+    try:
+        await controller.connect(
+            entry.data[CONF_HOST], entry.data[CONF_PORT], entry.data[CONF_PASSWORD]
+        )
+        await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)
+    except (OSError, InvallidPassword, LoadFailure) as err:
+        raise PlatformNotReady from err
```

Comment:
- This is not a platform, so we cannot raise a `PlatformNotReady` here.
- `InvallidPassword` is misspelled
- Invalid authentication should throw an `ConfigEntryAuthFailed` and have handling for it in the configuration flow.

Reply:
Invallid authentication is already handled in the config flow, but this can happen at this stage if the user chanegs the password on the node after they created the config entry.

Thats why we also catch it here